### DISCUSSION TITLE
Range Datepicker is not working as expected

### DISCUSSION THREAD

**Discussion by Sathyanarayana775** [carbon-design-system/carbon#12892] [Help]
Hi Team,
1. In one of our project, we are using the "Range DatePicker", We are receiving the data(just think, we are receiving the data from 1960 to 2020) from the API response and then we are setting the dates to date picker i,e is start date is 1960-01-01  and end date is 2020-01-01. I can see the dates in the date picker that is 1960-01-01, 2020-01-01. Now I selected the end date 16-jan-2003 and I have not selected start date(previosuly selected date:14-dec-1967), when i click on the start date input field, dates calender will open and show start date. Please refer the screen shot.

<img width="215" alt="image" src="https://user-images.githubusercontent.com/121172373/208913832-58f5293f-82a3-469c-aa61-286a31a74800.png">

2. I have passed the dates(start date and end date) from the api response to the date picker when user landed to the page for the first time(not selected any dates from the date picker input field). Suppose if i select only end date and i have not selected the start date. Onchange event should throw the start and end date but i can see only end date but start date will not throw.

**Comment by Sathyanarayana775** [comment]
Hi Team,
Please guide me if anyone has any updates on this.

Thanks,
Sathya

**Reply by tw15egan** [reply]
It would be helpful if you could provide a reproduction via [stackblitz](https://stackblitz.com/github/carbon-design-system/carbon/tree/main/examples/vite?file=src%2FApp.jsx) for further debugging

**Reply by Sathyanarayana775** [reply]
@tw15egan 
I am trying to save my changes and send those changes via Stack blitz, but unfortunately, I am unable to save from side. If I save and open new tab i am loosing all the code.

**Reply by Sathyanarayana775** [reply]
Let me send you the code in the sand box
